test/test-parse-warning.cpp

```diff
@@ -327,6 +327,262 @@ TEST_F(test_parse_warning, warn_on_pointless_string_compare_literals) {
     EXPECT_THAT(p.errors, IsEmpty());
   }
 }
+
+TEST_F(test_parse_warning,
+       warn_on_pointless_non_strict_compare_against_array_literal) {
+  {
+    test_parser p(u8"x == []"_sv);
+    p.parse_and_visit_expression();
+    EXPECT_THAT(p.errors, IsEmpty());
+  }
+  {
+    test_parser p(u8"x != []"_sv);
+    p.parse_and_visit_expression();
+    EXPECT_THAT(p.errors, IsEmpty());
+  }
+}
+
+TEST_F(test_parse_warning,
+       warn_on_pointless_non_strict_compare_against_class_literal) {
```

Comment:
Nit: I'd merge all the no-warn test cases together. Maybe call the merged test `non_strict_compare_against_literal_is_not_a_warning`. Then I'd loop over test data. See `test_parse_var.new_style_variables_cannot_be_named_let` in test/test-parse-var.cpp for example.